src/Microsoft.ML.TimeSeries/PredictionFunction.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.ML.TimeSeries
+{
+    public interface IStatefulRowToRowMapper : IRowToRowMapper
+    {
+    }
+
+    public interface IStatefulTransformer : ITransformer
```

Comment:
>IStatefulTransformer [](start = 21, length = 20)

So, *as far as I know* all of these interfaces are used only in this one assembly. We don't *want* them used anywhere else. Why are they or their implementations public?

Reply:
Your observation is correct, they should be made internal.

---
In reply to: [236482471](https://github.com/dotnet/machinelearning/pull/1727#discussion_r236482471) [](ancestors = 236482471)